miri-script/src/commands.rs

```diff
@@ -479,9 +492,61 @@ impl Command {
         // Then test, and let caller control flags.
         // Only in root project as `cargo-miri` has no tests.
         e.test(".", &flags)?;
+
+        if coverage {
+            Self::show_coverage_report(&e)?;
+        }
         Ok(())
     }
 
+    fn show_coverage_report(e: &MiriEnv) -> Result<()> {
+        let profraw_files: Vec<_> = Self::profraw_files(".")?;
+
+        let sysroot = cmd!(e.sh, "rustc --print sysroot").read()?;
+
+        let rustlib = Self::rustlib(sysroot)?;
+        let mut profdata_bin = rustlib.clone();
+        profdata_bin.push("llvm-profdata");
+
+        // Merge the profraw files
+        let profraw_files_cloned = profraw_files.iter();
+        cmd!(e.sh, "{profdata_bin} merge -sparse {profraw_files_cloned...} -o merged.profdata")
+            .quiet()
+            .run()?;
+
+        // Create the coverage report.
+        let miri_dir = e.miri_dir.as_os_str();
+        let mut cov_bin = rustlib.clone();
+        cov_bin.push("llvm-cov");
+        let suffix = if cfg!(windows) { ".exe" } else { "" };
+        cmd!(
+            e.sh,
+            "{cov_bin} report --instr-profile=merged.profdata --object {miri_dir}/target/debug/miri{suffix} --sources src/"
+        ).quiet().run()?;
+
+        // Delete artifacts.
+        cmd!(e.sh, "rm {profraw_files...}  merged.profdata").quiet().run()?;
+        Ok(())
+    }
+
+    fn rustlib(sysroot: String) -> Result<PathBuf> {
+        let mut pathbuf = PathBuf::from(sysroot);
+        pathbuf.push("lib");
+        pathbuf.push("rustlib");
+        pathbuf.push(rustc_version::version_meta()?.host);
+        pathbuf.push("bin");
+        Ok(pathbuf)
```

Comment:
This seems to duplicate logic in `MiriEnv::new`. Instead, please just store the `libdir` that is computed there in a field.

Reply:
ah I missed that! thanks!